service/src/main/java/tds/config/repositories/impl/ConfigRepositoryImpl.java

```diff
@@ -58,4 +65,128 @@ public ConfigRepositoryImpl(NamedParameterJdbcTemplate jdbcTemplate) {
 
         return clientSystemFlags;
     }
+
+    @Override
+    public Optional<ClientSystemMessage> findClientSystemMessage(String clientName, String messageKey, String language, String clientDefaultLanguage, String context, String subject, String grade) {
+        List<String> langauges = Arrays.asList(language, clientDefaultLanguage);
```

Comment:
May want to add a `checkNotNull(language)` from common to ensure language isn't null.  Up to you since this service can only be called via the controller.